### ISSUE TITLE
Feedback not showing on fullscreen MODAL page on Android

### ISSUE THREAD

**Issue by terreb** [EddyVerbruggen/nativescript-feedback#11]
Feedback doesn't show up on fullscreen modal page on Android.

nativescript CLI 3.2.1
tns-core-modules 3.2.0
tns-android 3.2.0
nativescript-feedback 1.0.6

Android 7.1

**Comment by EddyVerbruggen** [comment]
I don't think we can do anything about that as it's likely to be internal to the library we're wrapping here. Just to be sure: can you share a project that reproduces the issue?

**Comment by terreb** [comment]
Hi @EddyVerbruggen, here is the sample project:
https://github.com/terreb/feedback

Feedback shows up on the modal page on iOS but doesn't on Android where it's basically hidden behind the page. Does it do something with elevation on Android?

**Comment by terreb** [comment]
How do I get the feedback's nativeView?

**Comment by EddyVerbruggen** [comment]
Thanks for the sample, I can confirm the modal blocks the view.

The LoC responsible for adding the alert to the view [is this one](https://github.com/Tapadoo/Alerter/blob/4e913758a9d8f2059f5bdc9a13631e564e1d5f3d/alerter/src/main/java/com/tapadoo/alerter/Alerter.java#L143).

**Comment by terreb** [comment]
Ok, thanks, seems it's possible to get access to the native view by the reference
feedback.lastAlert, but only if it's already shown at least once. Setting elevation doesn't help. I'm not an Android developer, but another thing to try could be getting it's parent and then do bringChildToFront. I'm not sure whether the modal page and alerter are children of the same parent though. Would something like that work?

**Comment by EddyVerbruggen** [comment]
It's worth a shot.

**Comment by mhmo91** [comment]
Any updates here @EddyVerbruggen ?

**Comment by EddyVerbruggen** [comment]
No.

**Comment by petekanev** [comment]
Hey @EddyVerbruggen any idea if the plugin still works with the tns-core-modules changes after 4.0? No toasts seem to show, and I haven't had time to troubleshoot it.

**Comment by EddyVerbruggen** [comment]
@Pip3r4o I’m using it in this app which has all the latest bits: https://github.com/EddyVerbruggen/nativescript-pluginshowcase

Or are you referring to using it in a modal? I’ve never tried that.

**Comment by mhmo91** [comment]
@EddyVerbruggen if a modal is active n then the feedback is called, it shows behind the modal, so for me i had to close the modal to show the user the feedback, which is annoying..

**Comment by petekanev** [comment]
@mhmo91 that will supposedly be the native behavior too.

**Comment by webleaf** [comment]
need this too

**Comment by EddyVerbruggen** [comment]
I can't find a way to make it work. Leaving this open for others to take a stab.

**Comment by sanved77** [comment]
When the modal works, it displays something like - 
`DecorView: onWindowFocusChangedFromViewRoot hasFocus: true, DecorView@9d71eda[NativeScriptActivity]`
It fails to log this on a fullscreen activity. 
I believe this might just be because of the Modal isn't given the permission to draw on the top of the root view.

**Comment by sanved77** [comment]
Just some food for thought, I personally had it not working too, until I realized that I am opening up the modal in an **async await** making it run in a background thread instead of the UI thread, thus making it impossible for it draw on the UI thread. 
And when it tries to draw it on the background thread, the DecorView doesn't get the View to draw on and throws an exception which is being caught by the JS. 
You can try and set this to false under package.json to experiment - `discardUncaughtJsExceptions`

**Comment by jerbob92** [comment]
@sanved77 all the Javascript in NS runs on the UI thread. Only things running in a worker don't run in on the UI thread.
I'll try to get this to work tomorrow.

**Comment by jerbob92** [comment]
I made a fix to at least be able to make the notification appear when in a dialog. 
The solution is not pretty, it's in #62 . It might be useful for some. I will try to see if I can fix the issues.

**Comment by FlawaCLV** [comment]
@jerbob92 I've tried your solution on my project but still not working... It looks like the `android: { 
addToView: ... }` is not taken into account.

EDIT: ok my bad, I've looked at your commit and the plugin has to be updated first. Thanks.

**Comment by MrSnoozles** [comment]
@EddyVerbruggen @jerbob92 
https://github.com/Tapadoo/Alerter/releases/tag/7.0.0 was released 17 days ago. The changelog says "Added dialog support".
Would this fix this issue?

**Comment by jerbob92** [comment]
> @EddyVerbruggen @jerbob92
> https://github.com/Tapadoo/Alerter/releases/tag/7.0.0 was released 17 days ago. The changelog says "Added dialog support".
> Would this fix this issue?

Yes, that does sound like it will fix it. You could try it out :)

**Comment by MrSnoozles** [comment]
I could and would like to, but I don't know how to write NativeScript plugins that use native components 😢

**Comment by jerbob92** [comment]
@MrSnoozles I don't think you have to develop anything. You can probably add something to your app gradle file to force the new version of the native component:

```
  configurations.all {
    resolutionStrategy.force "com.tapadoo.android:alerter:7.0.1"
  }
  ```

**Comment by jerbob92** [comment]
@MrSnoozles After looking at the changes of the project we might need to make some changes to support showing in a dialog indeed. We probably still need to detect whether we are in a modal like I put in this MR:
https://github.com/EddyVerbruggen/nativescript-feedback/pull/62

We then need to pass the dialog into the constructor of the Alerter.
